src/utils/HealthLogger.js

```diff
@@ -165,6 +166,92 @@ define(function (require, exports, module) {
             fileEncCountMap[encoding]++;
             setHealthData(healthData);
         }
+
+
+        sendAnalyticsData("usagefileOpen" + language._name,
+                            "usage",
+                            "fileOpen",
+                            language._name.toLowerCase()
+                         );
+
+    }
+
+    /**
+     * Whenever a file is saved call this function.
+     * The function will send the analytics Data
+     * We only log the standard filetypes and fileSize
+     * @param {String} filePath The path of the file to be registered
+     */
+    function fileSaved(docToSave) {
+        if (!docToSave) {
+            return;
+        }
+        var fileType = docToSave.language ? docToSave.language._name : "";
```

Comment:
@SnchitGrover Do we want an empty string here or something like "unknown"? Empty strings are sometimes treated as null values while serialization depending on platform. Depends on your use case.

Reply:
@niteskum Is there ever a possibility of not having the `language` property in a document object?

Reply:
@shubhsnov  I think document will always have language. but since file save is critical operation , don't want to leave any space in code to trigger exception so I am doing  null check.